pkg/config/store/interface.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+)
+
+// Store represents a persistence backend for config.Config
+type Store interface {
```

Comment:
I am not sure about this name. We ought to use names that represent the object and not the action. 
The store shows an interface to work with the configuration of the plugins. Why not a better name here such as PluginConfig?

Reply:
`Store` is also a noun and is inpired on `React` and similar frameworks where a `Store` is used to abstract the persistence layer for some state (in our case `Config` is the state we are persisting). It would be the equivalent to how `Factory` abstracts the creation of new objects.

This said, if a better name is presented I would change this. However, `PluginConfig` or `PluginDataConfig` are not good names as this is the persistence layer of the project configuration, it has no direct relation with plugins (plugin configuration is also stored inside the project configuration but thats not all what it is stored there).

Reply:
Thank you for the explanation. I did not like too much the approach. However, it is fine. It addresses our requirements and we can also review that in the future if/when required. thank you for the explations.